tools/proofers/codepoint_validator.js

```diff
@@ -34,24 +34,103 @@ $(function () {
+
+    // In checking glyphs with a combining mark there will be only a few valid
+    // combinations. First find candidate pairs, then check if valid.
+    // This version checks only one combining mark but could be extended.
+    // Multiple marks and marks at start of lines are removed first.
+    // Do two passes: first single codepoints, then combining pairs.
+    // If marking or removing, first pass must let through all combining marks
+    // and second pass find all pairs (character + mark).
+    // If only testing for bad characters, the first pass can allow through
+    // only valid combining marks and second pass can find only pairs with
+    // these marks before validating.
+    // If testing a glyph 2nd pass can just test for pairs of codepoints.
+
+    // With unicode escapes we can use \p{Mn} instead of bmpCombiners
+
+    var processBadPattern = new RegExp("[^" + bmpCombiners + charClass + "]", "ug");
+
+    // any char followed by a combining char
+    var ProcessCombiningPattern = new RegExp(".[" + bmpCombiners + "]", "ug");
```

Comment:
`ProcessCombiningPattern` should be `processCombiningPattern` to match the other naming convention.